src/libraries/Microsoft.PowerFx.Core/Public/ExpressionError.cs

```diff
@@ -170,4 +170,21 @@ internal static IEnumerable<ExpressionError> New(IEnumerable<IDocumentError> err
             }
         }
     }
+
+    /// <summary>
+    /// Used to compare CheckResult.Errors and avoid duplicates.
+    /// </summary>
+    internal class ExpressionErrorComparer : EqualityComparer<ExpressionError>
+    {
+        // We compare only Message
+        public override bool Equals(ExpressionError error1, ExpressionError error2)
+        {
+            return error1.ToString() == error2.ToString();
```

Comment:
>error1.ToString() == error2.ToString(); [](http://example.com/codeflow?start=19&length=39)

if error1 and/or error2 is/are null, this will fail
consider adding 

if (error1 == null && error2 == null)
return true;

if (error1 ==null || error2 == null)
return false;

then return error1.ToString...